lua/entities/gmod_wire_expression2/core/vector2.lua

```diff
@@ -1096,8 +1096,8 @@ __e2setcost(7)
 -- Returns a random vector4 between -1 and 1
 e2function vector4 randvec4()
 	local vec = { random()*2-1, random()*2-1, random()*2-1, random()*2-1 }
-	local length = sqrt( vec[1]^2+vec[2]^2+vec[3]^2+vec[4]^2 )
-	return { vec[1] / length, vec[2] / length }
+	local length = ( vec[1]^2+vec[2]^2+vec[3]^2+vec[4]^2 ) ^ 0.5 -- x ^ 0.5 <=> math.sqrt( x )
```

Comment:
I tested it inside gmod and sqrt was exactly the same as ^0.5, while
math.sqrt is much much slower due to having to access the global scope to
find the function. The reason this commit did this (changed sqrt to ^0.5)
is because everywhere else in the file it was already done by someone else.

(god damnit wrong account. deleted the other comment and replied via email.
you'll all probably get two emails now derp)

On Tue, Mar 29, 2016 at 11:54 AM, Abigail <notifications@github.com> wrote:

> In lua/entities/gmod_wire_expression2/core/vector2.lua
> <https://github.com/wiremod/wire/pull/1111#discussion_r57697953>:
>
> > @@ -1096,8 +1096,8 @@ __e2setcost(7)
> >  -- Returns a random vector4 between -1 and 1
> >  e2function vector4 randvec4()
> >  	local vec = { random()*2-1, random()*2-1, random()*2-1, random()*2-1 }
> > -	local length = sqrt( vec[1]^2+vec[2]^2+vec[3]^2+vec[4]^2 )
> > -	return { vec[1] / length, vec[2] / length }
> > +	local length = ( vec[1]^2+vec[2]^2+vec[3]^2+vec[4]^2 ) ^ 0.5 -- x ^ 0.5 <=> math.sqrt( x )
>
> I've just tested on Lua 5.3.2 and LuaJIT 2.0 - in pure Lua, math.sqrt
> beats x ^ 0.5. In LuaJIT, sqrt beats x ^ 0.5 which beats math.sqrt.
>
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> <https://github.com/wiremod/wire/pull/1111/files/eea4bd4ab566561a9f22eae97efc8752ce2b6ae7#r57697953>
>